### PR TITLE
Ghost roles: Outpost Survivor and Syndicate Fugitive

### PR DESCRIPTION
**Pull request by JustinTrotter**
## About the PR

Added two new ghost roles to appear on salvage expedition missions. Probability can be adjusted via CCvars.
 
Outpost Survivor - essentially a passenger role that starts on a salvage planet. Current probability is 10% chance for the role to spawn on each salvage mission with breathable atmosphere. Main reason they are here is to add an element uncertainty for the expedition crew, so they don't automatically assume the survivor is a fugitive.

Syndicate Fugitive - a traitor role with only an objective to escape on the shuttle uncuffed. Current probability is 10% of spawned Outpost Survivors making them pretty damn rare. They start in prisoner jumpsuits with no equipment except an emergency crowbar (in case the power is off on the outpost or they spawn inside a glass garden). To make matters worse for them, as soon as the ghost role is requested centcom will automatically send wanted notices to all fax machines on the station outing them by name as a fugitive. Since there is no direct communication during expeditions, the fugitive has some leeway to bluff their way onto the shuttle. As soon as they FTL to the station however, they either need to hope nobody noticed the faxes or they gotta flee immediately.

**Media**

ghostrolefugitive
fugitivewantednotice
fugitivesurvivorinv
fugitivesurvivor
fugitiveobjectives

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: There have been reported sightings of stranded civilians on some salvage expedition missions. Please keep an eye out for them.

### CODE REVIEW

**Comment by brainfood1183** [comment]
I think these 2 roles should look identical from a salvs perspective so they can never be sure if its a survivor or a syndie.

**Comment by JustinTrotter** [comment]
> I think these 2 roles should look identical from a salvs perspective so they can never be sure if its a survivor or a syndie.

Is there currently a way to randomize the jumpsuit and shoes someone spawns with? I want the fugitives to start with a prisoner jumpsuit so that the first thing they need to do is hide/destroy it and find other clothes available on the outpost. This was if the fugitive is fairly sloppy and salv with a keen eye could discover the clothes.

File: Resources/Locale/en-US/fugitive/fugitive.ftl
Comment (Equivocateur) [inline Resources/Locale/en-US/fugitive/fugitive.ftl:2]:
```suggestion
fugitive-wanted-notice-description-line1 = Subject is wanted for crimes commited against Nanotrasen including but not limited to: Grand Theft, Bribery, and Sabotage.
```

File: Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl
Comment (Equivocateur) [inline Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:150]:
```suggestion
ghost-role-information-outpost-survivor-description = You are the last one alive on an overrun outpost. Try to survive until help arrives.
```

File: Resources/Locale/en-US/fugitive/fugitive.ftl
Comment (Equivocateur) [inline Resources/Locale/en-US/fugitive/fugitive.ftl:3]:
```suggestion
fugitive-wanted-notice-description-line2 = CAUTION: May be armed and dangerous. Security officers are to arrest on sight.
```

**Comment by JustinTrotter** [comment]
> > I think these 2 roles should look identical from a salvs perspective so they can never be sure if its a survivor or a syndie.
> 
> Is there currently a way to randomize the jumpsuit and shoes someone spawns with? I want the fugitives to start with a prisoner jumpsuit so that the first thing they need to do is hide/destroy it and find other clothes available on the outpost. That way if the fugitive is fairly sloppy and salv with a keen eye could discover the clothes.
> 
> Randomizing the survivors spawned jumpsuit would help with lowering suspicions.

Eh. Randomizing the jumpsuits is getting dangerously out of scope. I'll do it in another PR later. For now fugitives are starting with Outpost Survivor gear.

**Review by Equivocateur — CHANGES_REQUESTED** [review]
grammar

File: Resources/Locale/en-US/fugitive/fugitive.ftl
Comment (Equivocateur) [inline Resources/Locale/en-US/fugitive/fugitive.ftl:2]:
```suggestion
fugitive-wanted-notice-description-line1 = Subject is wanted for crimes commited against Nanotrasen including but not limited to: Grand Theft, Bribery, and Sabotage.
```

Comment (Equivocateur) [reply Resources/Locale/en-US/fugitive/fugitive.ftl:2]: Almost there

**Review by Equivocateur — CHANGES_REQUESTED** [review]

**Comment by Myakot** [comment]
I love this, however isn't instantly sending the fax upon ghost spawn way too harsh of a condition? In order to avoid instant arrest the agent would have to square against a heavily armoured salv squad, get a suit and somehow bunker down until shuttle arrives, since no TK for stealthy items. Otherwise he'll just be delivered straight to Sec and get cuffed. (although some people would love to play as prisoners from roundstart, so maybe it's fine)

If it's possible - I think there should be a timer on fax being sent to the station.

**Comment by JustinTrotter** [comment]
> I love this, however isn't instantly sending the fax upon ghost spawn way too harsh of a condition? In order to avoid instant arrest the agent would have to square against a heavily armoured salv squad, get a suit and somehow bunker down until shuttle arrives, since no TK for stealthy items. Otherwise he'll just be delivered straight to Sec and get cuffed. (although some people would love to play as prisoners from roundstart, so maybe it's fine)
> 
> 
> 
> If it's possible - I think there should be a timer on fax being sent to the station.

Yeah. In response to your comments, I had a quick discussion with a friend of mine and we agree with you. The solution we've come up with is to delay the wanted notice by a randomish time (between 2 - 10 minutes) after the expedition mission has ended. I may still base the timer on when the role is accepted and adjust the timer by a flat 15 minutes (typical mission time) for simplicity.

**Comment by JoeHammad1844** [comment]
sec would probably powergame this and just camp salv when they see the fax, id still love to see this added however

**Comment by JustinTrotter** [comment]
This PR has been open for about a month waiting for reviews from maintainers. I get that there's an antag freeze due to code quality issues with antag PRs lately which I'm cool with.

`Antagonists. We have had code quality issues with these and the code will be scrutinised much more heavily than normal.`

Can I please get some feedback? I don't want to have to repeatedly resolve conflicts if this is never going to get looked at.

**Comment by metalgearsloth** [comment]
> Can I please get some feedback? I don't want to have to repeatedly resolve conflicts if this is never going to get looked at.

Reposting my comment from #codebase-changes.

Just a reminder that if your PR gets conflicts you do not need to resolve them immediately. We get 700+ PRs a month and your PR is bound to get conflicts. You only need to resolve them if a maintainer asks you to. Most conflicts are just import conflicts which are trivial and a waste of time to resolve constantly.

File: Content.Server/Fugitive/FugitiveSystem.cs
Comment (metalgearsloth) [inline Content.Server/Fugitive/FugitiveSystem.cs:68]: Don't use async like this because it won't work for several reasons.
1. Doesn't work if server restarts.
2. Doesn't work if game mode changes.
3. Doesn't work if game is paused.

Don't use cvars for stuff that should be on the gamerule.

Haven't looked at the rest yet.

**Comment by UbaserB** [comment]
The idea is super cool, but the balancing feels a bit off to me tbh. Can’t wait to see how it turns out

**Comment by AJCM-git** [comment]
@JustinTrotter you coming back to this?

**Comment by JustinTrotter** [comment]
> @JustinTrotter you coming back to this?

Yes. Just need to find time.

**Comment by metalgearsloth** [comment]
re-open when you've had a chance to look

**Pull request closed without merging**
